src/vellum/workflows/monitoring/tests/workflow.py

```diff
@@ -0,0 +1,137 @@
+"""Simple workflow for testing monitoring decorators - Production-style setup."""
+
+import inspect
+from typing import Any, List, Type
+
+from vellum.workflows import BaseWorkflow
+from vellum.workflows.inputs.base import BaseInputs
+from vellum.workflows.monitoring.decorators import MonitorDecorator
+from vellum.workflows.nodes import MapNode
+from vellum.workflows.nodes.bases import BaseNode
+from vellum.workflows.outputs import BaseOutputs
+from vellum.workflows.state import BaseState
+
+
+class Inputs(BaseInputs):
+    fruits: List[str]
+
+
+class Iteration(BaseNode):
+    item = MapNode.SubworkflowInputs.item
+    index = MapNode.SubworkflowInputs.index
+
+    class Outputs(BaseOutputs):
+        count: int
+
+    def run(self) -> Outputs:
+        return self.Outputs(count=len(self.item) + self.index)
+
+
+class IterationSubworkflow(BaseWorkflow[MapNode.SubworkflowInputs, BaseState]):
+    graph = Iteration
+
+    class Outputs(BaseOutputs):
+        count = Iteration.Outputs.count
+
+
+class MapFruitsNode(MapNode):
+    items = Inputs.fruits
+    subworkflow = IterationSubworkflow
+
+
+class SimpleMapExample(BaseWorkflow[Inputs, BaseState]):
+    graph = MapFruitsNode
+    _monitoring_initialized = False  # Class-level flag for the entire workflow system
+
+    class Outputs(BaseOutputs):
+        final_value = MapFruitsNode.Outputs.count
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        # Production-style: Top-level workflow initializes monitoring for all components
+        if not SimpleMapExample._monitoring_initialized:
```

Comment:
[nit] `_monitoring_initialized = True` skips monitoring for my workflow which is a little unintuitive. Would expect opt-in

Reply:
making it opt-in also allows us to add it to BaseWorkflow